misc/bazel/pkg.bzl

```diff
@@ -0,0 +1,37 @@
+load("@rules_pkg//pkg:providers.bzl", "PackageFilegroupInfo", "PackageFilesInfo")
+
+def _pkg_overlay_impl(ctx):
+    destinations = {}
+    files = []
+    depsets = []
+
+    for src in reversed(ctx.attr.srcs):
+        pfi = src[PackageFilesInfo]
+        dest_src_map = {k: v for k, v in pfi.dest_src_map.items() if k not in destinations}
+        destinations.update({k: True for k in dest_src_map})
+        if dest_src_map:
+            new_pfi = PackageFilesInfo(
+                dest_src_map = dest_src_map,
+                attributes = pfi.attributes,
+            )
+            files.append((new_pfi, src.label))
+            depsets.append(depset(dest_src_map.values()))
+    return [
+        PackageFilegroupInfo(
+            pkg_files = reversed(files),
```

Comment:
Can you explain why we need the `reversed` here and in `DefaultInfo`? To me, it seems like we're constructing `files` and `depsets` to be free of duplicates, and therefore order shouldn't matter?

Reply:
The `reversed` here and below are not strictly required for soundness, but I think they make the end result less surprising: the order in which files will be written in the `rules_pkg` manifest and then installed will follow the order in which things have been added, and similarly with the list of files printed by bazel taken from `DefaultInfo`